src/Console/Migrate.php

```diff
@@ -57,24 +142,51 @@ protected function migrateTable(string $table, string $column): void
             });
     }
 
+    /**  if primary key is not string then skipping existing records */
+    protected function isNumericColumn($table, $column)
+    {
+        return $this->getPrimaryKeyType($this->source(), $table, $column) !== 'string';
+    }
+
     protected function tables(): array
     {
-        return $this->source->getAllTables();
+        if (!empty($this->table_names)) {
+            return $this->table_names;
+        }
+
+        return  $this->retrive_tables_from_target
+            ? $this->target->getAllTables()
+            : $this->source->getAllTables();
     }
 
     protected function cleanTargetDatabase(): void
     {
+        if (!$this->drop_target_tables) {
+            return;
+        }
+
         $this->info('Clearing the target database...');
 
         $this->target->dropAllTables();
     }
 
     protected function runMigrations(): void
     {
+        $runMigrationOn = $this->getMigrationOption();
+
+        if ($runMigrationOn === self::$none) {
+            return;
+        }
+
         $this->info('Run migrations on the databases...');
 
-        $this->call('migrate', ['--database' => $this->source()]);
-        $this->call('migrate', ['--database' => $this->target()]);
+        if ($runMigrationOn === self::$both_connection) {
+            $this->call('migrate', ['--database' => $this->source()]);
```

Comment:
Since you are given the choice of running migrations, you need to get rid of code duplication.